rosetta/text/vw_helpers.py

```diff
@@ -71,47 +73,65 @@ def parse_lda_topics(topics_file, num_topics, normalize=True):
         The --readable_model output of a VW lda run
     num_topics : Integer
         The number of topics in every valid row
+    max_token_hash : Integer
+        Reading of token probabilities from the topics_file will ignore all
+        token with hash above this value. Useful, when you know the max hash
+        value of your tokens.
     normalize : Boolean
-        Normalize the rows so that they represent probabilities of topic
-        given hash_val
+        Normalize the rows of the data frame so that they represent
+        probabilities of topic given hash_val. Note: must have get_iter=False.
+    get_iter : Boolean
+        if True will return a iterator yielding dict of hash and token vals
 
     Notes
     -----
     The trick is dealing with lack of a marker for the information printed
     on top, and the inconsistant delimiter choice.
     """
+    topics_iter = _parse_lda_topics_iter(topics_file=topics_file,
+                                         num_topics=num_topics,
+                                         max_token_hash=max_token_hash)
+    assert not (get_iter and normalize), ('get_iter must be set to False to ' +
+                                          'return normalized results')
+    if get_iter:
+        return topics_iter
+    else:
+        topics = [t for t in topics_iter]
+        topics = pd.DataFrame.from_records(topics, index='hash_val')
+        if normalize:
+            topics = topics.div(topics.sum(axis=1), axis=0)
+        return topics
+
+
+def _parse_lda_topics_iter(topics_file, num_topics, max_token_hash=1e+100):
     fmt = 'topic_%0' + str(len(str(num_topics))) + 'd'
-    topics = {fmt % i: [] for i in range(num_topics)}
-    topics['hash_val'] = []
     # The topics file contains a bunch of informational printout stuff at
     # the top.  Figure out what line this ends on
     with smart_open(topics_file, 'r') as open_file:
         # Once we detect that we're in the valid rows, there better not be
         # any exceptions!
         in_valid_rows = False
-        for line in open_file:
+        for i, line in enumerate(open_file):
             try:
                 # If this row raises an exception, then it isn't a valid row
                 # Sometimes trailing space...that's the reason for split()
                 # rather than csv.reader or a direct pandas read.
+                topic_item = {}
                 split_line = line.split()
                 hash_val = int(split_line[0])
+                if hash_val > max_token_hash:
```

Comment:
I am not an expert in how this works... but why would you assume your hashes are in order instead of keeping a counter of the size of the hash table and comparing to that count?